src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClientBuilderExtensions.cs

```diff
@@ -16,15 +17,19 @@ public static class FunctionInvokingChatClientBuilderExtensions
     /// </summary>
     /// <remarks>This works by adding an instance of <see cref="FunctionInvokingChatClient"/> with default options.</remarks>
     /// <param name="builder">The <see cref="ChatClientBuilder"/> being used to build the chat pipeline.</param>
+    /// <param name="loggerFactory">An optional <see cref="ILoggerFactory"/> to use to create a logger for logging function invocations.</param>
     /// <param name="configure">An optional callback that can be used to configure the <see cref="FunctionInvokingChatClient"/> instance.</param>
     /// <returns>The supplied <paramref name="builder"/>.</returns>
-    public static ChatClientBuilder UseFunctionInvocation(this ChatClientBuilder builder, Action<FunctionInvokingChatClient>? configure = null)
+    public static ChatClientBuilder UseFunctionInvocation(
+        this ChatClientBuilder builder,
+        ILoggerFactory? loggerFactory = null,
```

Comment:
Making the developer pass in an `ILoggerFactory` manually is a little unexpected. I would have thought we'd pull it from DI, given we already have the `services` below. Having to pass it in manually will be pretty cumbersome as you'll need to call `GetRequiredService` or similar with this call.

Could we use `services.GetService()` here to get `ILoggerFactory` as an optional dependency, and use it if available? It's pretty common throughout ASP.NET Core to log stuff liberally - we don't need the app dev's explicit buy-in to do that. As long as it's logged at a low level (e.g., Debug or Trace) then it doesn't bother anyone who doesn't need to see the logs.

Reply:
I intended to pull it from DI if this was null, but obviously neglected to. I'll fix that.